axelor-business-project/src/main/java/com/axelor/apps/businessproject/listener/ProjectTaskListener.java

```diff
@@ -0,0 +1,43 @@
+package com.axelor.apps.businessproject.listener;
+
+import com.axelor.apps.project.db.ProjectTask;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.persistence.PreUpdate;
+
+public class ProjectTaskListener {
+  @PreUpdate
+  public void onProjectTaskPreUpdate(ProjectTask projectTask) {
+    calculateParentTaskProgress(projectTask);
+  }
+
+  public void calculateParentTaskProgress(ProjectTask projectTask) {
+    ProjectTask parentTask = projectTask.getParentTask();
+    if (parentTask != null) {
+      List<ProjectTask> childProjectTasks = parentTask.getProjectTaskList();
+      childProjectTasks =
+          childProjectTasks.stream()
+              .map(task -> Objects.equals(task.getId(), projectTask.getId()) ? projectTask : task)
```

Comment:
You can directly call the objects instead of the ids, the equals will also compare the ID of both objects.